environs/instances/instance.go

```diff
@@ -20,7 +20,7 @@ type Instance interface {
 
 	// Addresses returns a list of hostnames or ip addresses
 	// associated with the instance.
-	Addresses(context.ProviderCallContext) ([]corenetwork.Address, error)
+	Addresses(context.ProviderCallContext) (corenetwork.ProviderAddresses, error)
```

Comment:
This interface change is reflected in all the specific providers. There is no change in logic; they just are just changed to return the new type.